File: src/main/java/se/johannesdahlgren/util/Direction.java
package se.johannesdahlgren.util;

import java.util.Arrays;

public enum Direction {
  UP("U"),
  DOWN("D"),
  RIGHT("R"),
  LEFT("L"),
  UP_LEFT("UL"),
  UP_RIGHT("UR"),
  DOWN_LEFT("DL"),
  DOWN_RIGHT("DR");

  private final String shortName;

  Direction(String shortName) {
    this.shortName = shortName;
  }

  public static Direction fromShortName(String shortName) {
    return Arrays.stream(Direction.values())
        .filter(d -> d.shortName.equals(shortName))
        .findFirst()
        .orElseThrow(() -> new RuntimeException("Not a valid Direction"));
  }
}

File: src/main/kotlin/se/johannesdahlgren/aoc22/util/Direction.kt
package se.johannesdahlgren.aoc22.util

enum class Direction(val short: String) {
    UP("U"), DOWN("D"), RIGHT("R"), LEFT("L"), UP_LEFT("UL"), UP_RIGHT("UR");

    companion object {
        fun fromString(value: String): Direction {
            return entries.first { value == it.short }
        }
    }
}